x-pack/legacy/plugins/siem/public/components/drag_and_drop/drag_drop_context_wrapper.tsx

```diff
@@ -59,22 +59,25 @@ const onDragEndHandler = ({
  */
 export const DragDropContextWrapperComponent = React.memo<Props>(
   ({ browserFields, children, dataProviders, dispatch }) => {
-    function onDragEnd(result: DropResult) {
-      enableScrolling();
-
-      if (dataProviders != null) {
-        onDragEndHandler({
-          browserFields,
-          result,
-          dataProviders,
-          dispatch,
-        });
-      }
-
-      if (!draggableIsField(result)) {
-        document.body.classList.remove(IS_DRAGGING_CLASS_NAME);
-      }
-    }
+    const onDragEnd = useCallback(
+      (result: DropResult) => {
+        enableScrolling();
+
+        if (dataProviders != null) {
+          onDragEndHandler({
+            browserFields,
+            result,
+            dataProviders,
+            dispatch,
+          });
+        }
+
+        if (!draggableIsField(result)) {
+          document.body.classList.remove(IS_DRAGGING_CLASS_NAME);
+        }
+      },
+      [browserFields, dataProviders]
```

Comment:
We should probably enable the ['react-hooks/exhaustive-deps'](https://github.com/facebook/react/issues/14920) eslint rule to catch instances where we're not including all necessary dependencies. Looks like it's in https://github.com/elastic/kibana/blob/feceb0f98eb817f065834f8b6c9c628cee41383a/packages/eslint-config-kibana/javascript.js#L145

but removing a dependency doesn't show a warning for me. Sounds like it should work out of the box for `useEffect`/`useCallback`. Thoughts?